src/braket/simulator_v2/base_simulator_v2.py

```diff
@@ -1,123 +1,97 @@
+        ):
+            os.environ[k] = os.environ.get(k, default)
+        # install Julia and any packages as needed
+        os.environ["PYTHON_JULIAPKG_OFFLINE"] = "yes"
+        import juliacall
+
+        jl = juliacall.Main
+        jl.seval("using JSON3, BraketSimulator")
+        sv_stock_oq3 = """
+        OPENQASM 3.0;
+        input float theta;
+        qubit[2] q;
+        h q[0];
+        cnot q;
+        x q[0];
+        xx(theta) q;
+        yy(theta) q;
+        zz(theta) q;
+        #pragma braket result expectation z(q[0])
+        """
+        dm_stock_oq3 = """
+        OPENQASM 3.0;
+        input float theta;
+        qubit[2] q;
+        h q[0];
+        x q[0];
+        cnot q;
+        xx(theta) q;
+        yy(theta) q;
+        zz(theta) q;
+        #pragma braket result probability
+        """
+        r = jl.BraketSimulator.simulate(
+            "braket_sv_v2", sv_stock_oq3, '{"theta": 0.1}', 0
         )
-        py_result = str(result)
-        return py_result
-    except Exception as e:
-        _handle_julia_error(e)
-
-
-def translate_and_run_multiple(
-    device_id: str,
-    programs: Sequence[OpenQASMProgram],
-    shots: Optional[int] = 0,
-    inputs: Optional[Union[dict, Sequence[dict]]] = {},
-) -> List[str]:
-    jl = setup_julia()
-    irs = jl.Vector[jl.String]()
-    is_single_input = isinstance(inputs, dict) or len(inputs) == 1
-    py_inputs = {}
-    if (is_single_input and isinstance(inputs, dict)) or not is_single_input:
-        py_inputs = [inputs.copy() for _ in range(len(programs))]
-    elif is_single_input and not isinstance(inputs, dict):
-        py_inputs = [inputs[0].copy() for _ in range(len(programs))]
-    else:
-        py_inputs = inputs
-    jl_inputs = jl.Vector[jl.Dict[jl.String, jl.Any]]()
-    for p_ix, program in enumerate(programs):
-        irs.append(program.source)
-        if program.inputs:
-            jl_inputs.append(program.inputs | py_inputs[p_ix])
-        else:
-            jl_inputs.append(py_inputs[p_ix])
-
-    if device_id == "braket_sv_v2":
-        device = jl.BraketSimulator.StateVectorSimulator(0, 0)
-    elif device_id == "braket_dm_v2":
-        device = jl.BraketSimulator.DensityMatrixSimulator(0, 0)
-
-    try:
-        results = jl.BraketSimulator.simulate._jl_call_nogil(
-            device,
-            irs,
-            jl_inputs,
-            shots,
+        jl.JSON3.write(r)
+        r = jl.BraketSimulator.simulate(
+            "braket_dm_v2", dm_stock_oq3, '{"theta": 0.1}', 0
         )
-        py_results = [str(result) for result in results]
-    except Exception as e:
-        _handle_julia_error(e)
-    return py_results
+        jl.JSON3.write(r)
+        return jl
+
+
+def setup_pool():
+    global __JULIA_POOL__
+    __JULIA_POOL__ = Pool(processes=1)
+    __JULIA_POOL__.apply(setup_julia)
+    atexit.register(__JULIA_POOL__.join)
+    atexit.register(__JULIA_POOL__.close)
+    return
```

Comment:
is this thread safe? what if two Python threads call this at the same time? (i.e. should you wrap this in some kind of lock?)